DhsTranslator/src/IDhsAdapter.h

```diff
@@ -0,0 +1,277 @@
+        }
+        Keyword(const std::string &name, int8_t val) {
+            this->name = name;
+            type = DHS_DT_INT8;
+            value.intVal = val;
+        }
+        Keyword(const std::string &name, int16_t val) {
+            this->name = name;
+            type = DHS_DT_INT16;
+            value.intVal = val;
+        }
+        Keyword(const std::string &name, int32_t val) {
+            this->name = name;
+            type = DHS_DT_INT32;
+            value.intVal = val;
+        }
+        Keyword(const std::string &name, uint8_t val) {
+            this->name = name;
+            type = DHS_DT_UINT8;
+            value.intVal = val;
+        }
+        Keyword(const std::string &name, uint16_t val) {
+            this->name = name;
+            type = DHS_DT_UINT16;
+            value.intVal = val;
+        }
+        Keyword(const std::string &name, uint32_t val) {
+            this->name = name;
+            type = DHS_DT_UINT32;
+            value.intVal = val;
+        }
+        Keyword(const std::string &name, float val) {
+            this->name = name;
+            type = DHS_DT_FLOAT;
+            value.floatVal = val;
+        }
+        Keyword(const std::string &name, double val) {
+            this->name = name;
+            type = DHS_DT_DOUBLE;
+            value.doubleVal = val;
+        }
+        Keyword(const std::string &name, bool val) {
+            this->name = name;
+            type = DHS_DT_BOOLEAN;
+            value.boolVal = val;
+        }
+        Keyword(const std::string &name, const std::string &val) {
+            this->name = name;
+            type = DHS_DT_STRING;
+            value.stringVal = new std::string(val);
+        }
+        Keyword(const Keyword &other) {
+            name = other.name;
+            type = other.type;
+            if (type == DHS_DT_STRING) {
+                value.stringVal = new std::string(*other.value.stringVal);
+            } else {
+                value = other.value;
+            }
+        }
+        Keyword & operator=(const Keyword &other) {
+            name = other.name;
+            type = other.type;
+            if (type == DHS_DT_STRING) {
+                value.stringVal = new std::string(*other.value.stringVal);
+            } else {
+                value = other.value;
+            }
+            return *this;
+        }
+
+        ~Keyword() {
+            if (type == DHS_DT_STRING && value.stringVal != NULL) {
+                delete value.stringVal;
+            }
+        }
+        const std::string& getName() const {
+            return name;
+        }
+        DHS_DATA_TYPE getType() const {
+            return type;
+        }
+        template<typename T> T getValue() const {
+            throw(std::logic_error(
+                        "Attempt to retrieve unsupported type of value from Keyword object."));
+        }
+    private:
+        std::string name;
+        DHS_DATA_TYPE type;
+        union {
```

Comment:
Oh, man... I don't even remember the last time I saw a `union`.
I don't miss C++ programming one little bit.